app.js

```diff
@@ -33,7 +35,12 @@ function createApp({ createLogger = defaultCreateLogger } = {}) {
         })
     );
 
-    app.use(createLogger());
+    // Create the logger instance
+    const loggerMiddleware = createLogger();
+    // Use the middleware for request logging
+    app.use(loggerMiddleware);
+    // Attach the main logger instance to app.locals for global access
+    app.locals.logger = loggerMiddleware.logger;
```

Comment:
req.log already exists. From what I can see from this PR, you are using/passing a reference of the logger initial derived from req.log (added via `app.use(loggerMiddleware);`).

Where are you using app.locals.logger? Is this needed?